desktop/core/src/desktop/lib/botserver/views.py

```diff
@@ -91,7 +89,6 @@ def handle_on_message(channel_id, bot_id, text, user_id):
       if not response['ok']:
         raise PopupException(_("Error posting message"), detail=response["error"])
 
-@check_document_access_permission
```

Comment:
Removed the decorator because it was giving error. Looks like I was not doing correctly? @romainr

Reply:
Yes, the signature of the decorator is not corresponding. Have you checked its code?

In any case this is a great idea as we could ensure that only someone with access can actually unfurl it.

2 ways, just copy pasting the code for now or reusing the decorator by hacking it a bit:

```
def check_access(id):
  request = MockRequest(POST={user=user, 'notebook': id1})

  @check_document_access_permission
  def check_access(request): pass
  check_access()
```

better it probably to just 

copy paste its logic for now (we could extract it but this is too dangerous)

        if str(notebook_id).isdigit():
          document = Document2.objects.get(id=notebook_id)
          document.can_read_or_exception(request.user)
        else:
          Document2.objects.get_by_uuid(user=request.user, uuid=notebook_id)


Also obviously we would need some unit tests for that.

And cf. above was recommending to skip it in this PRs and open a follow-up one or it might drag for a while..

Reply:
Will start a followup PR for this